### DISCUSSION TITLE
Improving Web Search Performance

### DISCUSSION THREAD

**Discussion by spammenotinoz** [open-webui/open-webui#7402] [Q&A] [answered]
Hi Web Search is working reliably but it takes between 40 and 50 seconds to return results, when set to a result count of 4 and concurrent request 10.   Similar results with all search engines.
I have a perplexity clone using the same instance of Searxng and it takes around 3-4 seconds to respond, I am wondering what the difference is.
Is Open-WebUI visiting each of the web sites associated with the search results, rather than analysing the seach results?
Is Open-WebUI submitting the images to the LLM for processing or just the text?

Any recommendations to improve the speed?

**Comment by spammenotinoz** [comment]
No one else has this issue?

**Comment by spammenotinoz** [comment, marked as answer]
Unsure if it was just a co-incidence but seems repeatable on a fresh build.

Under documents, "Embedding Batch Size", mine was at 1.
Change to anything higher than search result count.  Perhaps embeddings are used in Web Search?

In my tests, changed "Embedding Batch Size" to 10 as my "Search Result Count" was 4.

Or this change, kicked something else into gear.

**Reply by mikygit** [reply]
There is no 'Embedding Batch Size' parameter unfortunately :-(
Would you have a screen shot to share?

**Reply by gelbphoenix** [reply]
> There is no 'Embedding Batch Size' parameter unfortunately :-( Would you have a screen shot to share?

It's at `/admin/settings/documents` under "Embedding".

**Comment by Carrerajorge** [comment]
¿Cómo se puede mejorar la búsqueda en la web? porque actualmente tarda entre 30 a 40 segundos una búsqueda.

Alguien mas tiene este problema?

**Comment by spammenotinoz** [comment]
From what I can tell search works similar to Perplexities Pro mode, not like Perplexity Standard or ChatGPT.
ie: The LLM doesn't just analyse the search results, it actually connects to the specified number of search results.  Possibly even adding them as embeddings (which could explain the fix in my case).
You could probably write a tool\function if you just wanted a standard perplexity style search.
Actually thinking about it, might look at writing a tool that when enabled, the prompt goes to the Perplexity Online API then to the chosen LLM, similar to a re-ranker.  Just an idea, but for search and speed difficult to beat the Perplexity Online API.

The thing that most people miss, is when you use tools like search the system prompt needs to be aware, to stop the LLM giving dumb responses.  I haven't quite got this right yet.

**Reply by mikygit** [reply]
"The thing that most people miss, is when you use tools like search the system prompt needs to be aware, to stop the LLM giving dumb responses. I haven't quite got this right yet."
Not sure i understood this statement, can you develop pls?

**Comment by kitaev-chen** [comment]
Me too. The performance problem makes web search function useless. I use Chat Nio for web search function.

**Comment by 1Shiroyuuki1** [comment]
I can only agree with @spammenotinoz. I'm using Google PSE and also tried using the web search with Vertex Gemini and GPT-4o — the result: about 50 - 120 seconds until the response appears in the web interface. I can't see any performance issues in terms of CPU or RAM, as the CPU load is below 5% at the time of the request, and with 64 GB of RAM, only about 4 GB are in use.

Prometheus + Grafana Monitoring during the requests:
CPU Load in %
RAM Load in GB
![image](https://github.com/user-attachments/assets/4d72bf01-1970-4086-a18b-b33b6e5671c9)

**Reply by spammenotinoz** [reply]
> I can only agree with @spammenotinoz. I'm using Google PSE and also tried using the web search with Vertex Gemini and GPT-4o — the result: about 50 - 120 seconds until the response appears in the web interface. I can't see any performance issues in terms of CPU or RAM, as the CPU load is below 5% at the time of the request, and with 64 GB of RAM, only about 4 GB are in use.
> 
> Prometheus + Grafana Monitoring during the requests: CPU Load in % RAM Load in GB ![image](https://private-user-images.githubusercontent.com/72495090/442842283-4d72bf01-1970-4086-a18b-b33b6e5671c9.png?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3NDg1NzQ5OTUsIm5iZiI6MTc0ODU3NDY5NSwicGF0aCI6Ii83MjQ5NTA5MC80NDI4NDIyODMtNGQ3MmJmMDEtMTk3MC00MDg2LWExOGItYjMzYjZlNTY3MWM5LnBuZz9YLUFtei1BbGdvcml0aG09QVdTNC1ITUFDLVNIQTI1NiZYLUFtei1DcmVkZW50aWFsPUFLSUFWQ09EWUxTQTUzUFFLNFpBJTJGMjAyNTA1MzAlMkZ1cy1lYXN0LTElMkZzMyUyRmF3czRfcmVxdWVzdCZYLUFtei1EYXRlPTIwMjUwNTMwVDAzMTEzNVomWC1BbXotRXhwaXJlcz0zMDAmWC1BbXotU2lnbmF0dXJlPWZiMTcxMTg4Njk4MDVjY2U1ZmMzNDQ2YTNhMjZlY2U2ZGY1NzgxZGQ2ZDkyNmJkZTgwNDQ2ZWYwZDkyMTQ5YjYmWC1BbXotU2lnbmVkSGVhZGVycz1ob3N0In0.GDOWvqTUyk05aMaB1JR7r5HFHyH9b7ON4TMEXWaDDu8)

My searches are almost instant now, I enabled by-pass embeddings.  Sure token usage is higher, but the speed and quality is massive.  I disable web search on the expensive models.
The other win is ChromaDB doesn’t grow with each web search….:)

**Reply by TheRealAlexV** [reply]
weird, even with embedding bypass turned on for both search and documents, im still getting 50-90 second searches.

**Reply by flan7** [reply]
> weird, even with embedding bypass turned on for both search and documents, im still getting 50-90 second searches.

I am also seeing the same issue

**Reply by qubit999** [reply]
same

**Comment by Alaric323** [comment]
I'm trying to look into this myself, using Qwen3-embedding.  It searches without much issue and in a timely manner (less than 2 seconds on ddgs, set to duckduckgo, using 5 sources).  The main delay appears to come from storing and retrieving from an embeddings collection.
